### ISSUE TITLE
[gcc] -Wtype-limits warning in voting.cpp

### ISSUE THREAD

**Issue by SwooshyCueb** [irods/irods#5804]
When compiling with GCC 11, the compiler generates a `-Wtype-limits` warning in `irods/server/core/src/voting.cpp`

```
irods/server/core/src/voting.cpp: In function ‘auto irods::experimental::resource::voting::{anonymous}::replica_exceeds_resource_free_space(irods::experimental::resource::voting::{anonymous}::context&)’:
irods/server/core/src/voting.cpp:97:23: warning: comparison of unsigned expression in ‘< 0’ is always false [-Wtype-limits]
   97 |         if (file_size < 0) {
      |             ~~~~~~~~~~^~~
```

Related issue: #4083

**Comment by SwooshyCueb** [comment]
Looks like `file_size` is a `rodsLong_t` now, which is signed.

**Comment by trel** [comment]
For posterity... #7154 was included in 4.3.1.
